postgresql-plugin/src/main/java/io/cdap/plugin/postgres/PostgresConnector.java

```diff
@@ -112,13 +114,25 @@ public StructuredRecord transform(LongWritable longWritable, PostgresDBRecord re
   }
 
   @Override
-  protected String getTableQuery(String database, String schema, String table) {
-    return String.format("SELECT * FROM \"%s\".\"%s\"", schema, table);
+  protected String getTableName(String database, String schema, String table) {
+    return String.format("\"%s\".\"%s\"", schema, table);
   }
 
   @Override
-  protected String getTableQuery(String database, String schema, String table, int limit) {
-    return String.format("SELECT * FROM \"%s\".\"%s\" LIMIT %d", schema, table, limit);
+  protected String getTableQuery(String database, String schema, String table, int limit, String sampleType,
```

Comment:
looks like this general structure is repeated a bunch. It would be better to refactor things to remove the duplicate code. Seems like we only need to get the query string from each individual plugin

Reply:
Adin has refactored this. Thank you for the feedback!